File: katas/8 kyu/even-or-odd/solution.py
# even-or-odd
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def even_or_odd(number): return 'Even' if number%2 == 0 else 'Odd'
# ------------------------------ CLEVER SOLUTION ----------------------------- #
def even_or_odd(number):
    return 'Odd' if number % 2 else 'Even'
# ----------------------------------- TEST ----------------------------------- #
test.assert_equals(even_or_odd(2), "Even")
test.assert_equals(even_or_odd(1), "Odd")
test.assert_equals(even_or_odd(0), "Even")
test.assert_equals(even_or_odd(1545452), "Even")
test.assert_equals(even_or_odd(7), "Odd")
test.assert_equals(even_or_odd(78), "Even")
test.assert_equals(even_or_odd(17), "Odd")
test.assert_equals(even_or_odd(74156741), "Odd")
test.assert_equals(even_or_odd(100000), "Even")
test.assert_equals(even_or_odd(-123), "Odd")
test.assert_equals(even_or_odd(-456), "Even")

File: katas/8 kyu/even-or-odd/solution.js
// even-or-odd
/* ------------------------------- MY SOLUTION ------------------------------ */
function even_or_odd(number) {
  return number%2 == 0 ? 'Even' : 'Odd'
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
/* ---------------------------------- TEST ---------------------------------- */
// const chai = require("chai");
// const Test = chai.assert.strictEqual; // replace Test.assertEqual with Test
// chai.config.truncateThreshold=0;

const chai = require('chai');
const assert = chai.assert;

describe("Sample tests",() => {
  
  it("2 is even", () => {
    assert.strictEqual(even_or_odd(2), "Even");
  });
  it("7 is odd", () => {
    assert.strictEqual(even_or_odd(7), "Odd");
  });
  it("-42 is even", () => {
    assert.strictEqual(even_or_odd(-42), "Even");
  });
  it("-7 is odd", () => {
    assert.strictEqual(even_or_odd(-7), "Odd");
  });
  it("0 is even", () => {
    assert.strictEqual(even_or_odd(0), "Even");
  });
});